modules/abstract-eth/src/abstractEthLikeNewCoins.ts

```diff
@@ -1902,56 +1902,79 @@ export abstract class AbstractEthLikeNewCoins extends AbstractEthLikeCoin {
+        backupPrivateOrPublicKeyShare,
+        gasPrice,
+        gasLimit,
+        nonce,
+        params.eip1559,
+        params.replayProtectionOptions
+      );
+    } else if (this.isGG18SigningMaterial(userPublicOrPrivateKeyShare, params.walletPassphrase)) {
+      const [userKeyCombined, backupKeyCombined] = this.getKeyCombinedFromTssKeyShares(
+        userPublicOrPrivateKeyShare,
+        backupPrivateOrPublicKeyShare,
+        params.walletPassphrase
+      );
+      const backupKeyPair = new KeyPairLib({ pub: backupKeyCombined.xShare.y });
+      const baseAddress = backupKeyPair.getAddress();
 
-    const backupKeyNonce = await this.getAddressNonce(backupKeyAddress);
+      let { tx } = await this.buildTssRecoveryTxn(baseAddress, gasPrice, gasLimit, params);
 
-    // get balance of backupKey to ensure funds are available to pay fees
-    const backupKeyBalance = await this.queryAddressBalance(backupKeyAddress);
+      const signableHex = tx.getMessageToSign(false).toString('hex');
+      if (!userKeyCombined || !backupKeyCombined) {
+        throw new Error('Missing key combined shares for user or backup');
+      }
+      const signature = await this.signRecoveryTSS(userKeyCombined, backupKeyCombined, signableHex);
+      const ethCommmon = AbstractEthLikeNewCoins.getEthLikeCommon(params.eip1559, params.replayProtectionOptions);
+      tx = this.getSignedTxFromSignature(ethCommmon, tx, signature);
 
-    const totalGasNeeded = gasPrice.mul(gasLimit);
-    const weiToGwei = 10 ** 9;
-    if (backupKeyBalance.lt(totalGasNeeded)) {
-      throw new Error(
-        `Backup key address ${backupKeyAddress} has balance ${(backupKeyBalance / weiToGwei).toString()} Gwei.` +
-          `This address must have a balance of at least ${(totalGasNeeded / weiToGwei).toString()}` +
-          ` Gwei to perform recoveries. Try sending some ETH to this address then retry.`
-      );
+      return {
+        id: addHexPrefix(tx.hash().toString('hex')),
+        tx: addHexPrefix(tx.serialize().toString('hex')),
+      };
+    } else {
+      // DKLS
+      throw new Error('DKLS recovery is not implemented yet');
     }
+  }
 
-    // get balance of wallet and deduct fees to get transaction amount, wallet contract address acts as base address for tss?
-    const txAmount = backupKeyBalance.sub(totalGasNeeded);
+  private isGG18SigningMaterial(keyShare: string, walletPassphrase: string | undefined): boolean {
+    const prv = this.bitgo.decrypt({
+      input: keyShare,
+      password: walletPassphrase,
+    });
+    try {
+      const signingMaterial = JSON.parse(prv);
+      return (
+        signingMaterial.pShare &&
+        signingMaterial.bitgoNShare &&
+        (signingMaterial.userNShare || signingMaterial.backupNShare)
+      );
+    } catch (error) {
+      return false;
+    }
+  }
 
-    // build recipients object
+  private async buildTssRecoveryTxn(baseAddress: string, gasPrice: any, gasLimit: any, params: RecoverOptions) {
```

Comment:
NIT: we could type gasPrice and gaslimit

Reply:
will make this change along with the DKLS signing PR